autoPyTorch/utils/common.py

```diff
@@ -35,6 +46,25 @@ def __str__(self) -> str:
             self.name, self.supported_types, self.user_defined, self.dataset_property)
 
 
+class HyperparameterSearchSpace(NamedTuple):
+    """
+    A class that holds the search space for an individual hyperparameter.
```

Comment:
should it be search space updates?

Reply:
I didnt use the word update as this is used as a default also. Now all hyperparameter search spaces need to be defined this way and when we pass an update, the update class creates this search space and passes it to the get_hyperparameter_search_space method. see here https://github.com/ravinkohli/Auto-PyTorch/blob/ad43a947a5447c75b4ccd6bd44335fb96aa226d8/autoPyTorch/utils/hyperparameter_search_space_update.py#L57. What I am confused about is that, as we are calling get_hyperparameter_search_space for the whole hyperparameter search space of the component maybe it can be confusing to use HyperparameterSearchSpace also for individual hyperparameters. What do you think?